WebServer/pages_response.py

```diff
@@ -0,0 +1,84 @@
+from http.cookies import SimpleCookie
+from urllib.parse import parse_qs
+
+
+def read_page(s, s_path):
+    try:
+        page_to_open = open(s_path[1:], 'r').read()
+    except OSError:
+        s.send_response(404)
+        page_to_open = open('error.html', 'r').read()
+    s.wfile.write(bytes(page_to_open, 'utf-8'))
+
+
+def hello_page(s):
+    s.send_response(200)
+    s.send_header('content-type', 'text/html')
+    s.end_headers()
+    read_page(s, '/hello_page.html')
+
+
+def authorize(s):
+    s.send_response(200)
+    s.send_header('content-type', 'text/html')
+    s.send_header('Set-Cookie', 'auth=1; HttpOnly')
+    s.end_headers()
+    read_page(s, '/auth.html')
+
+
+def de_authorize(s):
+    s.send_response(200)
+    s.send_header('content-type', 'text/html')
+    s.send_header('Set-Cookie', 'auth=0; HttpOnly')
+    s.end_headers()
+    read_page(s, '/deauth.html')
+
+
+def form(s):
+    s.send_response(200)
+    s.send_header('content-type', 'text/html')
+    s.end_headers()
+    cookie = s.headers.get('Cookie')
+    simple_cookie = SimpleCookie(cookie)
+    cookie_value = simple_cookie.get('auth')
+    s.wfile.write(bytes('<p><font color="#B2DDE1" face="impact" size="+3">form page</font></p>', 'utf-8'))
+    if cookie_value.value == '0' or cookie_value is None:
+        auth_page = '<p><a href="/auth.html"><font color="#899597" size="+1" face="verdana">AUTHORIZE</font></a></p>'
+        s.wfile.write(bytes(auth_page, 'utf-8'))
+        read_page(s, '/form.html')
+    else:
+        de_auth_page = '<p><a href="/deauth.html"><font color="#899597" size="+1" face="verdana">DE-AUTHORIZE' \
+                       '</font></a></p>'
+        s.wfile.write(bytes(de_auth_page, 'utf-8'))
+        read_page(s, '/form.html')
+
+
+def charge(s, body):
+    cookie = s.headers.get('Cookie')
+    simp_cookie = SimpleCookie(cookie)
+    cookie_value = simp_cookie.get('auth')
+    s.send_response(200)
+    s.end_headers()
+    if cookie_value.value == '1':
+        params = parse_qs(body.decode())
+        value = params['digit'][0]
+        s.wfile.write(bytes("""<br><p><center><font face="impact" size="+3" color="#93B1B4">
```

Comment:
Styling issue: the code doesn't look easily readable.
Advice: 
1. it's fine to use the `""" ... """` but do use it in multiline mode like:
```python 
res = """
   <br/><p><center><font face="impact" size="+3" color="#93B1B4">
   OK. Peter Parker had got %s$ when he had been in London.<br/> 
   I wonder if that was enought for him...
   </font></center></p>
""" % value
...use res...
``` 
 you can do that because html ignores redundant whitespaces.
2. also, it's better to have all html tags closed, even if they are self-closing: `<br/>` instead of `<br>`
3. also, it's becoming more and more common to use `"{}".format()` or `f"{}"` where possible instead of `"%s"`

Reply:
Corrected